### DISCUSSION TITLE
Post Decorator does not support Content Type application/x-ndjson

### DISCUSSION THREAD

**Discussion by flower-of-the-bridges** [mia-platform/community#161] [Q&A] [answered]
I'm trying to add a `post decorator` to the `/export` route of a single view.

When I'm deploying the configuration, the microservice gateway fails to start with the following log messages:

```
/home/node/app/lib/buildCallChains.js:100
      throw Error('Post decorators can\'t require the body of the response if the route-method couple allows response unknown content types')
      ^
Error: Post decorators can't require the body of the response if the route-method couple allows response unknown content types
    at buildPostCall (/home/node/app/lib/buildCallChains.js:100:13)
    at /home/node/app/lib/buildCallChains.js:257:31
    at Array.map (<anonymous>)
    at buildFunctionsChain (/home/node/app/lib/buildCallChains.js:257:15)
    at buildCallChains (/home/node/app/lib/buildCallChains.js:272:40)
    at afterEnv (/home/node/app/lib/router.js:155:22)
    at Plugin.exec (/home/node/app/node_modules/avvio/plugin.js:89:17)
    at Boot.loadPlugin (/home/node/app/node_modules/avvio/plugin.js:187:10)
    at Task.release (/home/node/app/node_modules/fastq/queue.js:147:16)
    at worked (/home/node/app/node_modules/fastq/queue.js:199:10)
```
It seems that the content-type of the `/export` endpoint of a crud resource is `application/x-ndjson`.
How can I add a Post Decorator having an `application/x-ndjson` as Content Type?

**Comment by davidebianchi** [comment, marked as answer]
Hi @giovf96, this error is due to the configuration and not the specific header. The microservice-gateway only support post decorator which receive body in JSON.

So, it is not possible through the microservice-gateway. You can always create a new endpoint which will call the export endpoint and handle all the request/response as stream (in this specific case, I suppose the nd-json is handled like a stream) and change the body accordingly with your needs.

**Reply by davidebianchi** [reply]
Oh, side node. If you want to change something which is not related to body, like response headers or maybe notify from the post decorator, you can do it. The only thing to do is to set the `requireResponseBody` params to false (if you use the console, there is a [specific checkbox](https://docs.mia-platform.eu/docs/development_suite/api-console/api-design/decorators#manage-your-decorators)).

**Reply by flower-of-the-bridges** [reply]
Hi @davidebianchi thanks for the suggestions!
